Changes for new backend checks

MancunianSam opened the pull request:
This lambda updates the API with the backend checks responses but instead of doing one at a time, it sends multiple updates. 
It will also send updates for redacted files and for the file and consignment statuses.
As this is no longer sending SQS messages and no longer encrypting environment variables (as there's nothing sensitive any more) there's a lot of deleted files.

TomJKing commented on src/main/resources/application.conf:
```diff
@@ -7,21 +7,9 @@ url {
     id = ${?CLIENT_ID}
-    secret = "secretplaceholder"
-    secret = ${?CLIENT_SECRET}
+    secret_path = "/intg/keycloak/backend_checks_client/secret"
```
Should this be pointing to the intg secret as a default?

MancunianSam replied:
It doesn't matter greatly but I've put it back to secretplaceholder.

TomJKing commented on src/main/scala/uk/gov/nationalarchives/api/update/Lambda.scala:
```diff
@@ -47,30 +46,79 @@ class Lambda {
+
+  def sendStatuses(input: Input, token: BearerAccessToken): Future[Statuses] = {
+    val (consignmentStatuses, fileStatuses) = input.statuses.statuses.partition(_.statusType == "Consignment")
+
```
Unnecessary line break

TomJKing commented on src/main/scala/uk/gov/nationalarchives/api/update/Lambda.scala:
```diff
@@ -47,30 +46,79 @@ class Lambda {
+    } yield {
+      output.write(Result(av, file, ffid, statuses).asJson.printWith(Printer.noSpaces).getBytes())
+    }
+    Await.result(result, 60.seconds)
```
Is there a reason for the `Await`?

MancunianSam replied:
Yeah, AWS lambda doesn't understand Scala futures, if you don't put the await in, it will terminate before it's had a chance to do anything.

TomJKing reviewed (commented):
Looks good. Just some minor comments.

Couple of general comments:

- noticed there are a couple of Jenkins files still in the project that should be deleted.
- could you update the README, its still referring to queues etc

MancunianSam commented:
I've deleted the Jenkinsfiles and removed the references to the queues from the README.
I've updated the deploy workflow to point to a new v2 version of the lambda which should stop this being a breaking change.

TomJKing reviewed (approved):
Looks good. Thanks
